core-bundle/src/Routing/ResponseContext/CoreResponseContextFactory.php

```diff
@@ -114,6 +121,35 @@ public function createContaoWebpageResponseContext(PageModel $pageModel): Respon
             )
         ;
 
+        if ($pageModel->enableCsp) {
```

Comment:
Split in a separate `private` method maybe to make things cleaner?

Reply:
Done in 4510a924f1